runelite-mixins/src/main/java/net/runelite/mixins/WorldHoppingMixin.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (c) 2018, Lotto <https://github.com/devLotto>
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package net.runelite.mixins;
+
+import net.runelite.api.mixins.Inject;
+import net.runelite.api.mixins.Mixin;
+import net.runelite.api.widgets.WidgetID;
+import net.runelite.rs.api.RSClient;
+
+@Mixin(RSClient.class)
+public abstract class WorldHoppingMixin implements RSClient
+{
+	@Inject
+	@Override
+	public void openWorldHopper()
+	{
+		// The clicked x & y coordinates (the last arguments) are not processed in the game or sent to Jagex, so they don't have to be real.
+		// Don't use a WidgetInfo, to prevent being able to pass in arbitrary group & child ids.
+		// Also don't use WidgetInfo.PACK to prevent being able to edit what that method returns to pass in an arbitrary packed widget id.
+		menuAction(-1, WidgetID.LOGOUT_PANEL_ID << 16 | WidgetID.LogoutPanel.WORLD_SWITCHER_BUTTON, 57, 1, "World Switcher", "", 658, 384);
```

Comment:
Generally we would access the WidgetInfo and `.getId()` here. See also `PACK`. Even if you get the group/child id off of the widgetinfo, that is preferable to this.